credsweeper/scanner/scan_type/pem_key_pattern.py

```diff
@@ -40,47 +50,73 @@ def run(cls, config: Config, rule: Rule, target: AnalysisTarget) -> Optional[Can
+        """Detects PEM key in single line and with iterative for next lines according
+        https://www.rfc-editor.org/rfc/rfc7468
 
         Args:
-            lines: Lines to be checked
+            config: Config
+            rule: Rule
+            target: Analysis target
 
         Return:
-            Boolean. True if PEM key, False otherwise
+            List of LineData with found PEM
 
         """
-        lines = cls.strip_lines(lines)
-        lines = cls.remove_leading_config_lines(lines)
+        line_data: List[LineData] = []
         key_data = ""
-        for line_num, line in enumerate(lines):
-            if line_num >= 190:
-                return False
-            if PEM_END_PATTERN in line:
-                # Check if entropy is high enough
-                removed_by_entropy = not Util.is_entropy_validate(key_data)
-                # Check if have no substring with 5 same consecutive characters (like 'AAAAA')
-                pattern_check = ValuePatternCheck(config)
-                removed_by_filter = pattern_check.equal_pattern_check(key_data)
-                not_removed = not (removed_by_entropy or removed_by_filter)
-                return not_removed
-            # PEM key line should not contain spaces or . (and especially not ...)
-            elif " " in line or "..." in line:
-                return False
-            else:
-                key_data += line
-
-        return False  # Return false if no `-END` section in lines
+        # get line with -----BEGIN which may contain full key
+        first_line = LineData(config, target.line, target.line_num, target.file_path, target.file_type, target.info,
+                              rule.patterns[0])
+        line_data.append(first_line)
+        # protection check for case when first line starts from 0
+        line_num = target.line_num if 0 < target.line_num else 1
+        finish_line = line_num + 200
+        for line in target.lines[line_num - 1:]:
+            if finish_line < line_num:
+                return []
+            if 1 != line_num and target.line_num != line_num:
+                _line = LineData(config, line, line_num, target.file_path, target.file_type, target.info,
+                                 cls.re_value_pem)
+                line_data.append(_line)
+            line_num += 1
+            # replace escaped line ends with real and process them - PEM does not contain '\' sign
+            sublines = line.replace("\\r", '\n').replace("\\n", '\n').splitlines()
+            for subline in sublines:
+                if cls.is_leading_config_line(subline):
+                    continue
+                elif PEM_END_PATTERN in subline:
+                    # Check if entropy is high enough for base64 set with padding sign
+                    entropy = Util.get_shannon_entropy(key_data, Chars.BASE64_CHARS.value)
+                    if 4.85 > entropy:
```

Comment:
Why it use `4.85` for compare?
As I know `Util.is_entropy_validate()` already has a standard `4.5`.
Please  check this.

https://github.com/Samsung/CredSweeper/blob/8afc05be9a6985be51770ef11615f0bc4df74b94/credsweeper/utils/util.py#L71-L82

Reply:
In benchmark minimal entropy of short key was 4.990... 
So, let it be 4.5